### DISCUSSION TITLE
4. Tuning window, the number of time-step, for behavior classifier (or pattern recognition)

### DISCUSSION THREAD

**Discussion by hyunjimoon** [Data4DM/BayesSD#11] [perspectives; sense-making]
Yaman's team named the window length as segment and set this as 12 from the [code](https://github.com/Data4DM/BayesSD/blob/97593d10a4b4755758a1425a8df15da4c9546568/Code_Conti/4_Auto/GeomPrior/aria/aria/models/ists.py#L235) (ISTS)). 
```
Pseudo-code of the classifier by Kanar (1999)
Receive the dynamic behavior, X, to be classified
Normalize X, and obtain Xnorm
Split Xnorm into equal-length segments, and obtain S,, Sz, ... S,
Iterate over all segments {
Pick a segment S;
Calculate mean for the segment, i.e., a;
Calculate slope for the segment, i.e., b;
Calculate curvature for the segment, i.e., c
Iterate over all defined behavior classes (
Pick a behavior class, B
Calculate the state-optimized likelihood of norm belonging to the B, class,
i.e., Pi
}
Calculate the maximum likelihood, Pmax=max(p)
Report the behavior class j for which the p=P max
```

**Comment by tomfid** [comment]
I think the model time horizon definitely influences at least our perception of behavior, or our ability to discern what is going on.
![image](https://user-images.githubusercontent.com/2151849/189975644-6c3964c1-947c-4084-9f6f-6a7201a8cbef.png)

Sin wave or logistic growth? What if there's measurement error?
![image](https://user-images.githubusercontent.com/2151849/189975686-1d9d6f80-dc90-4b9d-814e-28a4105ed01e.png)

I think a closely related challenge is the selection of a window for behavior classification. One could, for example, further decompose the trajectories above into 2 phases dominated by + and - feedback.

I'm guessing these questions are addressed in the BATS papers, but it's been a long time since I read them.

**Comment by hyunjimoon** [comment]
Similar spirit: Ridge or Multiple peak

![image](https://user-images.githubusercontent.com/30194633/195796543-a5407c4d-1661-4fc2-a711-bf4bee81379d.png)
I am not saying ridge and multiple local peaks are the same. However, our inspection can be the same. e.g. you can get the exact same samples (3,4,5) from two different distributions: gaussian mixture (.5 Normal(3, 1) + .5 Normal(5,1)) and Uniform (3,5).  First has two local peaks, the second is a ridge.

> Highly correlated parameters (e.g. your example) are not the same as multiple local peaks, though when you have them you get a ridge and you should essentially remove one parameter from optimization (i.e. fix/assume it). Local peaks are best identified by looking at convergence of multiple restarts in optimization to different locations/peaks.
